neo4j-to-neptune/src/main/java/com/amazonaws/services/neptune/util/NeptuneBulkLoader.java

```diff
@@ -219,117 +225,168 @@ protected CompletableFuture<Boolean> uploadFileAsync(String directoryPath, Strin
         System.err.println("Starting sequential upload of files from " +
             directoryPath + " to s3://" + bucketName + "/" + s3Prefix);
 
-        // Get all CSV files in the directory
-        File[] csvFiles = directory.listFiles((dir, name) ->
-            name.toLowerCase().endsWith(".csv") || name.toLowerCase().endsWith(".gz"));
+        // Get all files in the directory with the specified extension
+        File[] csvFiles = directory.listFiles((dir, name) -> name.toLowerCase().endsWith(".csv"));
 
         if (csvFiles == null || csvFiles.length == 0) {
             System.err.println("No files with correct extension were found in " + directoryPath);
-            return CompletableFuture.completedFuture(false);
+            throw new RuntimeException("No CSV files found in directory: " + directoryPath);
         }
 
-        // Sort files to upload vertices first, then edges (for better organization)
-        java.util.Arrays.sort(csvFiles, (a, b) -> {
-            if (a.getName().contains("vertices") && b.getName().contains("edges")) return -1;
-            if (a.getName().contains("edges") && b.getName().contains("vertices")) return 1;
-            return a.getName().compareTo(b.getName());
-        });
-
         // Upload files sequentially to avoid connection pool exhaustion
-        return uploadFilesSequentially(csvFiles, s3Prefix, 0);
+        uploadFilesSequentially(csvFiles, s3Prefix);
+        System.err.println("Successfully uploaded all " + csvFiles.length + " files from " + directoryPath);
     }
 
     /**
      * Upload files sequentially (one at a time) to avoid overwhelming the connection pool
      */
-    private CompletableFuture<Boolean> uploadFilesSequentially(File[] files, String s3Prefix, int index) {
-        if (index >= files.length) {
-            System.err.println("Successfully uploaded all " + files.length + " files sequentially");
-            return CompletableFuture.completedFuture(true);
-        }
-
-        File currentFile = files[index];
-        String csvFilePath = s3Prefix + "/" + currentFile.getName();
+    private void uploadFilesSequentially(File[] files, String s3Prefix) throws Exception {
```

Comment:
Nit: no checked exception thrown

```suggestion
    private void uploadFilesSequentially(File[] files, String s3Prefix) {
```